galley/pkg/kube/converter/converter.go

```diff
@@ -206,6 +216,39 @@ func kubeIngressResource(cfg *Config, _ resource.Info, name resource.FullName, _
 	return []Entry{e}, nil
 }
 
+func kubeServiceResource(_ *Config, _ resource.Info, name resource.FullName, _ string, u *unstructured.Unstructured) ([]Entry, error) {
+	var service corev1.Service
+	if err := convertJSON(u, &service); err != nil {
```

Comment:
Is this conversion always 1-1? If not, it maybe better to model this similar to how ingress => other resource conversion is also done. (see pkg/runtime for that).

Reply:
AFAICT, the conversion is N:1 (or even N:N). The [pilot k8s service registry code](https://github.com/istio/istio/blob/master/pilot/pkg/serviceregistry/kube/controller.go) is looking k8s endpoints (for populating endpoint data) and pod/nodes for AZ information.

I suspect this would look similar to how we [project ingress onto virtual service and gateway
](https://github.com/istio/istio/blob/e390f0af9d68c5dfbc55fdedcf09c4752a3f087d/galley/pkg/runtime/state.go#L147).

Reply:
We need to make it simple.

Galley: The conversion from k8s service to ServiceEntry is 1:1, so only one host will be populated in this case.

Pilot: ServiceEntry is mergeable, so if a new Endpoint is created, ideally we should use incremental MCP to add to the list of existing Endpoints.

Reply:
As long as this is 1:1 (and will always be), this is an ok place to do this. However, if there needs to be cross-resource lookups or merges, the right place is the projection code that @ayj is pointing at.

Reply:
I mean is 1:1 between a k8s service and a synthetic ServiceEntry.

 @ayj were you thinking of N:1 with Endpoints maybe?

There will be cross-resource lookups to populate Endpoints for sure. I was envisioning a simple get all Endpoints for service name.namespace would be sufficient, similar to pilot code, it's unclear what the projections do.

Reply:
Agreed with Oz to leave it in converter for now, will move to projections once we add endpoints (though at the moment seems like projections don't have access to labels so connecting endpoints with services might be a problem?)